ci/L0_multi_gpu/multi_lora/test.sh

```diff
@@ -62,7 +62,8 @@ model_json=$(cat <<EOF
     "enforce_eager": "true",
     "enable_lora": "true",
     "max_lora_rank": 32,
-    "lora_extra_vocab_size": 256
+    "lora_extra_vocab_size": 256,
+    "distributed_executor_backend":"ray"
```

Comment:
> For python native multiprocessing mode and KIND_MODEL setting, Triton hits "failed to stop server: Internal - Exit timeout expired.

Few questions on this:
1. Do we know more details or limitations on why this is happening?
2. Is this an error happening on server shutdown?
    - Is there some issue with the `python native multiprocessing` due to the details of Triton's python backend launching each instance as a separate process? 
3. Is this with 1, 2, or any amount of model instances with KIND_MODEL?

Reply:
> Do we know more details or limitations on why this is happening?

Issue with unclear multi-gpu test fail when upgrading to 0.5.0 versions and up
In PR [#5230](https://github.com/vllm-project/vllm/pull/5230) vllm changed default executor for distributed serving from Ray to python native multiprocessing for single node processing. This becomes in issue starting with v0.5.1 release.
For `python native multiprocessing` mode and KIND_MODEL setting, triton hits "failed to stop server: Internal - Exit timeout expired. Exiting immediately." and pt_main_thread processes are never stopped/killed.
Solution: add `"distributed_executor_backend":"ray"`  to `model.json`

> Is this an error happening on server shutdown? 

Yes, and I have a reproducer outside of triton

> Is this with 1, 2, or any amount of model instances with KIND_MODEL?

If "distributed_executor_backend" field is not specified, than for tp>2 and distributed among a single node, than `MP` backend kicks in. However, I've noticed that even when tp=1 and "distributed_executor_backend" is specified in `model.json`, vllm will go through distributed serving even when tp=1. More on the slack channel for this behavior